index.src.html

```diff
@@ -318,13 +318,28 @@ <h4 id="examples-password-signin">Password-based Sign-in</h4>
               if (credential.type == "<a const>password</a>") {
                 fetch("https://example.com/loginEndpoint", { credentials: credential, method: "POST" })
                   .then(function (response) {
-                    // Notify the user that signin succeeded! Do amazing, signed-in things!
+                    if (/* |response| indicates a successful login */) {
+                      // Store the credential again, see note below.
+                      navigator.<a attribute>credentials</a>.<a idl lt="store()" for="CredentialsContainer">store</a>(credential);
+                      // Notify the user that signin succeeded! Do amazing, signed-in things!
+                    } else {
+                      // Insert some code here to show a basic login form.
+                    }
                   });
               } else {
                 // See <a href="#examples-federated-signin">the Federated Sign-in example</a>
               }
             });
       </pre>
+      Note: To ensure that credentials are persisted correctly, a website
+      should always tell the credential manager when a sign-in succeeded - even
+      if the credentials were just retrieved from
+      {{CredentialsContainer/get()}}. This ensures that PSL matched credentials
+      (for example credentials stored for <code>https://www.example.com</code>
+      but provided for <code>https://m.example.com</code>, see
+      [[#security-cross-origin-leakage]]) become eligible for unmediated
+      access. If the verbatim credentials have been persisted before, this
+      storing has no effect and is invisible to the user.
```

Comment:
WDYT of rephrasing this to emphasize the edge cases? Something like the following:

> Note: The `<a>credential chooser</a>` presented by the user agent could allow the user to choose credentials that aren't actually stored for the current origin. For instance, it might offer up credentials from `<code>https://example.com</code>` when signing into `<code>https://www.example.com</code>` (as described in `[[#security-cross-origin-leakage]]`), or it might allow a user to create a new credential on the fly (as described in the other patch I looked at a few minutes ago). Developers can deal gracefully with this uncertainty by calling `{{CredentialsContainer/store()}}` when credentials are successfully used: if the credentials aren't yet stored for the origin, the user will be given the opportunity to do so. If they are stored, the user won't be prompted.